src/graphics/image.rs

```diff
@@ -293,3 +336,24 @@ impl Drop for Image {
         unsafe { ffi::sfImage_destroy(self.image) }
     }
 }
+
+#[derive(Debug, Copy, Clone)]
+pub enum SetPixelError {
+    XTooLarge(u32, u32),
+    YTooLarge(u32, u32),
+}
+
+impl Error for SetPixelError {}
+
+impl fmt::Display for SetPixelError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::XTooLarge(passed_in, max) => {
+                write!(f, "x index out of bounds. x:{} max_x:{}", passed_in, max)
```

Comment:
Perhaps instead of `max_x`/`max_y`, perhaps `width` and `height` would make more sense.